packages/ow-db/src/index.ts

```diff
@@ -1,2 +1,20 @@
-export * from "@prisma/client"
-export * from ".prisma/client"
+import { Prisma, PrismaClient } from "@prisma/client"
+
+declare global {
+  // eslint-disable-next-line no-var
+  var prisma: PrismaClient | undefined
+}
+
+const prismaOptions: Prisma.PrismaClientOptions = {}
+if (!!process.env.NEXT_PUBLIC_DEBUG) prismaOptions.log = ["query", "error", "warn"]
+
+export const prisma = globalThis.prisma || new PrismaClient(prismaOptions)
+
+export const customPrisma = (options: Prisma.PrismaClientOptions) => new PrismaClient({ ...prismaOptions, ...options })
+
+if (process.env.NODE_ENV !== "production") {
+  globalThis.prisma = prisma
+}
+
+export type { Prisma, PrismaClient }
+export default prisma
```

Comment:
What is the rationale behind making it available as global variable? I'm guessing some SSR thing with Next maybe but?

Also maybe remove either default or named export so there's not two?

Reply:
Personally, I just looked at the example in [turobopack](https://github.com/vercel/turbo/blob/main/examples/with-prisma/packages/database/src/client.ts) example repo, and what is done in calcom. Looks like this is about ensuring there's a single instance of prisma in the application (Singleton pattern).

Reply:
I would guess that it is to avoid multiple connections perhaps from the same server

Reply:
I *don't think* that would matter since Node will cache the module ensuring only one prisma instance is created but again, unsure how that works with bunding/next ssr/etc

Reply:
Probably fine to leave it as-is though LGTM

Reply:
As you mentioned, it might be for Nextjs. I'm considering using prisma for NextAuth and auth session handling (kinda prefer redis, but well well).